polyply/src/nanoparticle_constructor.py

```diff
@@ -0,0 +1,354 @@
+            threshold = length / 2  # divide nanoparticle region into 2
+            for index, entry in enumerate(core_numpy_coords):
+                if entry[2] > minimum_threshold + threshold:
+                    core_top_values[index] = entry
+                else:
+                    core_bot_values[index] = entry
+            self.core_indices = [core_top_values, core_bot_values]
+
+    def _identify_attachment_index(self):
+        """
+        Find index of atoms that corresponds to the atom on the ligand
+        which you wish to bond with the ligand on the nanoparticle core
+        """
+        self.attachment_index = None
+        for index, entry in enumerate(self.ligand_block.atoms):
+            if entry["atomname"] == self.ligand_anchor:
+                attachment_index = index
+        return attachment_index
+
+    def run(self, core_mol):
+        """
+        we have two blocks representing the blocks of the NP core and
+        the ligand block. Hence, we wish to add links to the blocks
+        first add connection between the blocks
+        """
+        self._set_ff()
+        self.np_molecule = self.NP_block.to_molecule()
+        ligand_len = len(list(self.ligand_block.atoms))
+        # compute the max radius of the
+        core_size = len(list(self.NP_block))
+        # self.attachment_index = 5
+        self._identify_indices_for_core_attachment()  # generate indices we want to find for core attachment
+        attachment_index = self._identify_attachment_index()
+        # tag on surface of the core to attach with ligands
+        core_index_relevant = self.core_indices[
+            0
+        ]  # first entry of the identified core indices
+        # core_index_relevant_II = self.core_indices[
+        #    1
+        # ]  # second entry of the identified core indices
+        attachment_list = []
+        # find the indices that are equivalent to the atom
+        # on the ligand we wish to attach onto the NP
+        # based on the ligands_N, and then merge onto the
+        # NP core the main ligand blocks ligand_N number of times
+
+        for index in range(0, self.ligands_N):
+            ligand_index = index * ligand_len  # compute starting index for each ligand
+            print(ligand_index, core_size)
+            scaled_ligand_index = ligand_index + core_size
+            attachment_list.append(scaled_ligand_index)
+            # merge the relevant molecule
+            self.np_molecule.merge_molecule(self.ligand_block)
+
+        # create list of interactions for first batch of indices we have on the
+        # core surface
+        for index, index2 in enumerate(
+            list(core_index_relevant.keys())[: self.ligands_N]
+        ):
+            interaction = vermouth.molecule.Interaction(
+                atoms=(index2, attachment_list[index] + attachment_index),
+                parameters=["1", "0.0033", "50000"],
+                meta={},
+            )
```

Comment:
I understand that this is just a draft. I'll add a comment anyways ;) If you only add an edge between the ligand and the core nanoparticle using `molecule.add_edge`, we can in a later step run `ApplyLinks` to get the bonded interactions straight from the force-field files. This will make it easier to adopt them later. 

So my suggestion:

```
np_molecule.add_edge(index2, attachment_list[index] + attachment_index)
```
Then run ApplyLinks() from polyply.src.apply_links in line 251